### DISCUSSION TITLE
Partial lack of functionality behind reverse proxy.

### DISCUSSION THREAD

**Discussion by tetricky** [docmost/docmost#427] [Help / Questions] [answered]
I'm running docmost in a podman container, behind caddy as a reverse proxy.

All seemed to be functioning well....users could see other collaborating users, and access and space membership all work fine.

Then I tried to draw something (excalidraw, and draw.io, others untested). The drawing interface starts, I can edit (but no collaborators) and no ability to save the drawing. so clearly the client side browser is not able to write the edit back to the server. I don't really know what I'm looking for....reading discussions, issues, and of course the documentation, I get the sense that it may be a websocket issue.

I have found this error in the developer tools of the client browser

```
Object { stack: "dt@https://docs.example.com/assets/index-B3W_nTMv.js:67:47813\nNB@https://docs.example.com/assets/index-B3W_nTMv.js:69:1034\nv@https://docs.example.com/assets/index-B3W_nTMv.js:69:6081\nEventHandlerNonNull*wse</<@https://docs.example.com/assets/index-B3W_nTMv.js:69:6150\nwse<@https://docs.example.com/assets/index-B3W_nTMv.js:69:5480\nx9@https://docs.example.com/assets/index-B3W_nTMv.js:71:512\npromise callback*_request@https://docs.example.com/assets/index-B3W_nTMv.js:72:1079\nrequest@https://docs.example.com/assets/index-B3W_nTMv.js:71:1881\nn/<@https://docs.example.com/assets/index-B3W_nTMv.js:72:1671\nmB/<@https://docs.example.com/assets/index-B3W_nTMv.js:67:41945\nTd@https://docs.example.com/assets/index-B3W_nTMv.js:505:49381\nw@https://docs.example.com/assets/index-B3W_nTMv.js:884:21834\nasync*OJ@https://docs.example.com/assets/index-B3W_nTMv.js:37:9864\nRJ@https://docs.example.com/assets/index-B3W_nTMv.js:37:10018\nIJ@https://docs.example.com/assets/index-B3W_nTMv.js:37:10075\npO@https://docs.example.com/assets/index-B3W_nTMv.js:37:31517\nL$@https://docs.example.com/assets/index-B3W_nTMv.js:37:31936\nzS/<@https://docs.example.com/assets/index-B3W_nTMv.js:37:36865\nFT@https://docs.example.com/assets/index-B3W_nTMv.js:40:36983\no$@https://docs.example.com/assets/index-B3W_nTMv.js:37:8999\nzS@https://docs.example.com/assets/index-B3W_nTMv.js:37:33224\nmT@https://docs.example.com/assets/index-B3W_nTMv.js:37:17378\nYJ@https://docs.example.com/assets/index-B3W_nTMv.js:37:17160\nEventListener.handleEvent*j$@https://docs.example.com/assets/index-B3W_nTMv.js:37:32733\n$S@https://docs.example.com/assets/index-B3W_nTMv.js:37:32132\ntg/<@https://docs.example.com/assets/index-B3W_nTMv.js:37:32299\ntg@https://docs.example.com/assets/index-B3W_nTMv.js:37:32243\nnte@https://docs.example.com/assets/index-B3W_nTMv.js:40:19770\nDz@https://docs.example.com/assets/index-B3W_nTMv.js:40:40129\nIz@https://docs.example.com/assets/index-B3W_nTMv.js:40:39881\ncte@https://docs.example.com/assets/index-B3W_nTMv.js:40:39749\nZb@https://docs.example.com/assets/index-B3W_nTMv.js:40:39597\nOz@https://docs.example.com/assets/index-B3W_nTMv.js:40:34669\nC@https://docs.example.com/assets/index-B3W_nTMv.js:25:1560\n$@https://docs.example.com/assets/index-B3W_nTMv.js:25:1924\nEventHandlerNonNull*@https://docs.example.com/assets/index-B3W_nTMv.js:25:2090\n@https://docs.example.com/assets/index-B3W_nTMv.js:25:3826\n", message: "Request failed with status code 400", name: "AxiosError", code: "ERR_BAD_REQUEST", config: {…}, request: XMLHttpRequest, response: {…}, status: 400 }
```

The documentation is not clear about what is required, nor how to achieve it, to attain this functionality. I've had a look around and tried to add a websocket definition in my caddy file (which I totally do not understand and have probably made a right mess of), in the form:

```
@websockets {
	header Connection *Upgrade*
	header Upgrade websocket
}
reverse_proxy docmost_pod:3000
```

This hasn't worked.

I understand this is largely a result of my incomplete understanding, and the use behind a reverse proxy in general may not be supported (although I have seen using a reverse proxy is the recommended approach for https/TLS). I would however appreciate any detailed, or specific to caddy, guidance as to how to correctly serve docmost behind a reverse proxy.

As a general comment, docmost is a brilliant simple, yet comprehensive, markdown based documentation application. Very much appreciated. Personally I'd like SSO (lldap/authelia for oidc) and public viewing....but that would just be cherry on the top.

Thank you very much.

**Comment by Philipinho** [comment]
Can you upload attachments?

**Comment by tetricky** [comment]
No. It fails, and I appear to get a 400 error.

**Reply by Philipinho** [reply]
It’s a file permissions issue. 
Try the solution here https://github.com/docmost/docmost/issues/111.

**Comment by tetricky** [comment, marked as answer]
Yes. It absolutely appears to be that. I `chown 1000:1000` to the volume bind, and that now allows upload and saving in drawings.

Thank you very much.
